app/controllers/merchant_items_controller.rb

```diff
@@ -14,12 +14,17 @@ def show
   def edit
     @merchant = Merchant.find(params[:merchant_id])
     @item = Item.find(params[:id])
+
+    if params.has_key?(:status)
+      @item = @item.update(item_params)
```

Comment:
I think any time we're updating a database entry it should really be done through the #update action. In this case since you're just clicking a button to change it and not needing any input on the #edit view page, we can completely bypass the #edit action

Reply:
Complete.